Automattic/vip-design-system
# Add accessible tabs component

## Description
<img width="479" alt="Screenshot 2022-11-08 at 15 59 57" src="https://user-images.githubusercontent.com/668251/200598780-4d90b913-a149-4c45-bcec-6c95ff6e61c3.png">

Introducing a more accessible Tab component by using the Radix one https://www.radix-ui.com/docs/primitives/components/tabs

The new component is, for now `NewTabs`.
Since the sub-component don't clash with the older `TabItem` I kept the original names so when we finished the replacement we can only rename `NewTabs` in `Tabs`

Since radix also requires a tab `value` that is used as an index to link the tabs button with their content, I've used this `value` to add some classes to both the buttons and the content too. 

## Checklist


## Steps to Test

Outline the steps to test and verify the PR here.

Example:

1. Pull down PR.
1. `npm run dev`.
1. Open Storybook on the `NewTabs` component https://deploy-preview-130--vip-design-system-components.netlify.app/?path=/story/newtabs--default
1. Compare the visual style of the tabs [with the old one](https://deploy-preview-130--vip-design-system-components.netlify.app/?path=/story/tabs--default)
1. Verify that the tabs are accessible, that they are announced and you can move with the left/right arrow when focus is on them
